coderd/rbac/roles_test.go

```diff
@@ -630,6 +678,37 @@ func TestRolePermissions(t *testing.T) {
 				},
 			},
 		},
+		{
+			// Notification preferences are currently not organization-scoped
+			// Any owner/admin across any organization may access any users' preferences
+			// Members may access their own preferences
+			Name:     "NotificationPreferencesAnyOrg",
+			Actions:  []policy.Action{policy.ActionRead, policy.ActionUpdate},
+			Resource: rbac.ResourceNotificationPreference.AnyOrganization().WithOwner(currentUser.String()),
```

Comment:
Remove `AnyOrganization`. This resource is not org scoped, so it should not use this function.

Reply:
I thought I could use this precisely because there's no org scoping just yet; is this semantically incorrect?

Reply:
My understanding is that organizations shouldn't even enter into the equation if the resource isn't org-scoped and that adding them has the potential to confuse future readers.

Reply:
OK cool, I've removed the org-related tests entirely.

Reply:
@johnstcn is correct orgs should never enter the equation.

I should add more guards to `AnyOrganization`, as it should only really be used by the UI.

AnyOrganization still org scopes a resource. You can read the PR description about it here: https://github.com/coder/coder/pull/14009